tests/test_table.py

```diff
@@ -572,6 +590,42 @@ def test_to_hdf5_error(self):
             t.to_hdf5(h5, 'tests')
 
     @npt.dec.skipif(HAVE_H5PY is False, msg='H5PY is not installed')
+    def test_to_hdf5_custom_formatters(self):
+        self.st_rich = Table(self.vals,
+                             ['1', '2'], ['a', 'b'],
+                             [{'taxonomy': ['k__a', 'p__b']},
+                              {'taxonomy': ['k__a', 'p__c']}],
+                             [{'barcode': 'aatt'}, {'barcode': 'ttgg'}])
+
+        def bc_formatter(grp, category, md, compression):
+            name = 'metadata/%s' % category
+            data = np.array([m[category].upper() for m in md])
+            grp.create_dataset(name, shape=data.shape, dtype=H5PY_VLEN_STR,
+                               data=data, compression=compression)
+
+
+        fname = mktemp()
```

Comment:
I still think this should be `mkstemp`, since this function (`mktemp`) was deprecated almost 12 years ago. In the entire biom-format codebase, `mktemp` is used 5 times, all of them in this file (test_table.py). I'll create an issue to replace all `mktemp`s with `mkstemp`s so that it oesn't block this PR. See #616.